backend/echo/settings.py

```diff
@@ -0,0 +1,162 @@
+    'corsheaders',
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.gis',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'django_extensions',
+    'rest_framework',
+    'rest_framework_gis',
+    'api',
+]
+
+MIDDLEWARE = [
+    'corsheaders.middleware.CorsMiddleware',
+    'django.middleware.security.SecurityMiddleware',
+    'django.contrib.sessions.middleware.SessionMiddleware',
+    'django.middleware.common.CommonMiddleware',
+    'django.middleware.csrf.CsrfViewMiddleware',
+    'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'django.contrib.messages.middleware.MessageMiddleware',
+    'django.middleware.clickjacking.XFrameOptionsMiddleware',
+]
+
+ROOT_URLCONF = 'echo.urls'
+
+TEMPLATES = [
+    {
+        'BACKEND': 'django.template.backends.django.DjangoTemplates',
+        'DIRS': [],
+        'APP_DIRS': True,
+        'OPTIONS': {
+            'context_processors': [
+                'django.template.context_processors.debug',
+                'django.template.context_processors.request',
+                'django.contrib.auth.context_processors.auth',
+                'django.contrib.messages.context_processors.messages',
+            ],
+        },
+    },
+]
+
+WSGI_APPLICATION = 'echo.wsgi.application'
+
+
+# Database
+# https://docs.djangoproject.com/en/3.2/ref/settings/#databases
+
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.contrib.gis.db.backends.postgis',
+        'NAME': os.getenv('POSTGRES_DB'),
+        'USER': os.getenv('POSTGRES_USER'),
+        'PASSWORD': os.getenv('POSTGRES_PASSWORD'),
+        'HOST': os.getenv('POSTGRES_HOST'),
+        'PORT': os.getenv('POSTGRES_PORT')
+    }
+}
+
+
+# Password validation
+# https://docs.djangoproject.com/en/3.2/ref/settings/#auth-password-validators
+
+AUTH_PASSWORD_VALIDATORS = [
+    {
+        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
+    },
+    {
+        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
+    },
+    {
+        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
+    },
+    {
+        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
+    },
+]
+
+
+# Internationalization
+# https://docs.djangoproject.com/en/3.2/topics/i18n/
+
+LANGUAGE_CODE = 'en-us'
+
+TIME_ZONE = 'UTC'
+
+USE_I18N = True
+
+USE_L10N = True
+
+USE_TZ = True
+
+
+# Static files (CSS, JavaScript, Images)
+# https://docs.djangoproject.com/en/3.2/howto/static-files/
+
+STATIC_URL = '/static/'
+
+# Default primary key field type
+# https://docs.djangoproject.com/en/3.2/ref/settings/#default-auto-field
+
+DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
+
+WATCHMAN_ERROR_CODE = 503
+WATCHMAN_CHECKS = (
+    'watchman.checks.databases',
+    'api.checks.gazetteercache'
+)
+
+# CORS settings
+CORS_ORIGIN_WHITELIST = [
```

Comment:
Though it looks like OAR did something fancy here: https://github.com/open-apparel-registry/open-apparel-registry/blob/develop/src/django/oar/settings.py#L380-L399 -- If this whitelist won't suffice, I am happy to emulate what OAR does in the above linked code block.

Reply:
I think the differences with OAR extend beyond just the different CORS configuration. The CORS configuration on OAR is actually not relevant for the main OAR site, because OAR serves the static frontend and the API from the same host and port (e.g. while visiting `https://openapparel.org/` one might see a request to `https://openapparel.org/api/contributor-lists/?contributors=2422`). This means that CORS doesn't apply at all for the application's frontend because the API requests aren't cross-origin in the first place. I believe the fancier setup in OAR is to allow use of the OAR API from within certain other websites that have some kind of relationship with OAR, which isn't really something we expect to do.

The current setup here, by contrast, is cross-origin because the API and frontend are hosted on different ports, and if we continue that architecture through to deployment, we'll end up with the API hosted on something like `api.echosearch.org/api/` while the rest of the app is hosted at `www.echosearch.org`, and then we'll have to add `*.echosearch.org` to this whitelist.

I see a few different ways to proceed from here.

If we want to get closer to how OAR does things, OAR uses `django-spa` and `whitenoise` to serve the static files from within the Django app, and they use Webpack's proxy feature to pass through requests to `/api/` and `/admin/` through to the Django server. Mastarm supposedly [has a proxy feature](https://github.com/azavea/mastarm#build), but it's not clear how well it works, so it may be that to use proxying in the build tool we'd need to [migrate away from Mastarm](https://github.com/azavea/echo-locator/issues/376).

Alternatively, we could stick with the status quo and commit to making the app cross-origin for the foreseeable future. This would avoid the work of upgrading Mastarm, and would allow us to leave the existing deployment scheme for static files largely untouched. If we switched to hosting the files in Django, we'd need to swap out the existing S3-based hosting with Django static hosting for everything. However, this would potentially make the app more confusing for administrative users because the Django admin would have to be accessed at something like `api.echosearch.org/admin/`; if we put everything on the same domain it could be `www.echosearch.org/admin/` which seems friendlier.

Overall, Mastarm has been a continuing source of confusion and frustration (including one other thread in this PR), so I'm leaning toward the same-origin solution so that we have another excuse to get rid of it, but I could also see sticking with a cross-origin solution to avoid the need to rework the static deployment (even in an imaginary world where we were already using a more standard bundler).

So I think my suggested path forward would be to check out Mastarm's proxy feature and try to add `django-spa` if it seems to work. If that all works out, then we can eliminate `CORS` support entirely. If we run into roadblocks then I think we can leave this as-is and prioritize removing Mastarm. How does that sound to you?

Reply:
Yes, this plan sounds very good to me. I think the reasoning here makes a lot of sense. 

I do think we should start by testing if adding `django-spa` and `whitenoise` to serve static files and also update the `proxy` flag for `mastarm build` works. If this won't work as expected, I am with you on the fact that switching away to another bundler is a wise choice here since `mastarm` is not maintained and we now have a chance to make the switch since we need to make the local frontend dev work with a Django API. I do not think that the currently implemented `CORS` solution is a way to go considering the downstream build and infra overhead.

I did take a stab at adding `django-spa` and `whitenoise`, plus modifying the `--proxy` flag in [this commit](https://github.com/azavea/echo-locator/pull/419/commits/432b9113b09b55deb83ccd8d60d0759d4a228ea4), but I think I was missing something there and it did not work for me locally. I am happy to pair with you when you get a chance to have this figured out together!

Reply:
Update: discussed about the above on Slack, and it [this commit](https://github.com/azavea/echo-locator/pull/419/commits/c09e13776a1ea54109463fc12f4d7d19d14e175d) fixed the issue for local development 🎉